invenio/ext/login/__init__.py

```diff
@@ -151,11 +153,11 @@ def do_login_first(error=401):
         if not urllib.unquote(secure_url).startswith(request.base_url):
             return redirect(secure_url)
         if current_user.is_guest:
-            flash(g._("Please sign in to continue."), 'info')
+            if not session.get('_flashes'):
```

Comment:
Checks if any message have been flash, if not it flashes one. Not sure if this is the best way to do it.